### PR TITLE
Refactor s3 settings (move settings parsing into single place)

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>CI Settings</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step
- [ ]  Allow: Integration Tests
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Unit tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All with aarch64
- [ ]  Allow: All with ASAN
- [ ]  Allow: All with TSAN
- [ ]  Allow: All with Analyzer
- [ ]  Allow: All with Azure
- [ ]  Allow: Add your option here
---
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN
- [ ]  Exclude: All with MSAN
- [ ]  Exclude: All with UBSAN
- [ ]  Exclude: All with Coverage
- [ ]  Exclude: All with Aarch64
---
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)
- [ ]  allow: batch 1 for multi-batch jobs
- [ ]  allow: batch 2
- [ ]  allow: batch 3
- [ ]  allow: batch 4, 5 and 6
</details>

This was partially already done by https://github.com/ClickHouse/ClickHouse/pull/59767, but not fully.

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit a504622ee46988681d2aad050badcb9f4c3e5a6e with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64412/a504622ee46988681d2aad050badcb9f4c3e5a6e/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/IO/S3Common.h [hunk 1]
```diff
@@ -68,40 +66,140 @@ namespace Poco::Util
...
 };
 
-namespace DB::S3
+namespace DB
 {
+class NamedCollection;
+struct ProxyConfigurationResolver;
 
-HTTPHeaderEntries getHTTPHeaders(const std::string & config_elem, const Poco::Util::AbstractConfiguration & config);
+namespace S3
```
Comment (CheSema) [inline src/IO/S3Common.h:74]: I'm very happy to see this structured lists. Huge thank to you!

**Review by CheSema — APPROVED** [review]

**Comment by kssenii** [comment]
> AST fuzzer (debug) 

https://github.com/ClickHouse/ClickHouse/issues/65055

> Integration tests (aarch64) [5/6] — fail: 1

`test_checking_s3_blobs_paranoid/test.py::test_query_is_canceled_with_inf_retries` - fails for a while now in all PRs

> Stateless tests (debug) [4/5] — fail: 1

`00626_replace_partition_from_table_zookeeper` -- not related to changes

File: src/IO/S3Common.cpp [hunk 2]
```diff
@@ -98,104 +102,320 @@ ServerSideEncryptionKMSConfig getSSEKMSConfig(const std::string & config_elem, c
...
+    /// But it is good enough unless very high `burst` values are used.
+    if (UInt64 max_get_rps = isChanged("max_get_rps") ? get("max_get_rps").get<UInt64>() : settings.s3_max_get_rps)
+    {
+        size_t default_max_get_burst = settings.s3_max_get_burst
+            ? settings.s3_max_get_burst
+            : (Throttler::default_burst_seconds * max_get_rps);
 
-    if (from.no_sign_request.has_value())
-        no_sign_request = from.no_sign_request;
+        size_t max_get_burst = isChanged("max_get_burts") ? get("max_get_burst").get<UInt64>() : default_max_get_burst;
```
Comment (Algunenano) [inline src/IO/S3Common.cpp:302]: Noticed the typos while refactoring this (`max_get_burts` -> `max_get_burst`, `max_put_burts` -> `max_put_burst`).

Aside from the typos, I don't quite see how this is supposed to work. Who could populate RequestSettings::max_get_burts considering the setting is not declared inside this class?

**Pull request merged**